src/store.js

```diff
@@ -24,6 +24,31 @@ export const CONFIG_UPDATED = 'config.updated';
 export const EFFECTIVE_GENOME_UPDATED = 'effective.genome.updated';
 export const STORE_DESTROYED = 'store.destroyed';
 
+export function expKeyStatesHas(keyStates, stateName, key, prefix) {
+  prefix = prefix || false;
+
+  let hasIt = false;
+  keyStates.experiments.forEach(function(expKeyStates) {
+    const state = expKeyStates.get(stateName);
+    if (!state) {
+      return;
+    }
+
+    if (prefix) {
+      state.forEach(function(stateKey) {
+        if (!key || strings.startsWith(stateKey, key)) {
+          hasIt = true;
+        }
+      });
+    } else {
+      if (state.has(key)) {
+        hasIt = true;
+      }
+    }
+  });
+  return hasIt;
```

Comment:
You don't necessarily need to change:
I think the `some()` operator would be clearer
```suggestion
  return keyStates.experiments.some(function(expKeyStates) {
    const state = expKeyStates.get(stateName);
    if (!state) {
      return false;
    }

    if (prefix) {
      state.forEach(function(stateKey) {
        if (!key || strings.startsWith(stateKey, key)) {
          return true;
        }
      });
    } else {
      if (state.has(key)) {
        return true;
      }
    }
    
    return false;
  });
```